starter-code/routes/courses.js

```diff
@@ -0,0 +1,29 @@
+const express = require("express");
+const courses = express.Router();
+
+const { ensureLoggedIn, ensureLoggedOut } = require("../middlewares/auth");
+const passport = require("passport");
+
+const User = require("../models/user");
+const Courses = require("../models/course");
+
+function checkRoles(role) {
+  return function(req, res, next) {
+    if (req.user.role === role) {
+      return next();
+    } else {
+      res.redirect("/login");
+    }
+  };
+}
+
+const checkTA = checkRoles("TA");
+
+courses.get("/", checkTA, (req, res) => {
```

Comment:
unless you append a url when you call "app.use" (i.e. app.use("/courses")) this will be a double handler for you root url